cmd/machine-api-operator/main.go

```diff
@@ -24,6 +27,8 @@ var (
 )
 
 func init() {
+	osconfigv1.AddToScheme(scheme.Scheme)
```

Comment:
Outside of tests where it's a quick-fix, I'm not sure `init()` is the right place for this. Shouldn't it happen in `operator.New()` or something?

Reply:
`init()` is called only once, `operator.New` can be called multiple times. Imho, `AddToSheme` will fail if the same type is registered twice.